### DISCUSSION TITLE
Work around for using proc-macro libs

### DISCUSSION THREAD

**Discussion by Christian-Prather** [corrosion-rs/corrosion#569] [Q&A] [answered]
Hello,
I am currently evaluating using Corrosion to simplify some existing cmake magic in a project. The general structure is that there is a top level cmake and Cargo.toml file which defines a workspace of packages. One of these packages is a macro library which leverages proc-macro.

Unfortunately since Corrosion only supports staticlib, cdylib, and dylib AND because proc-macro libs cannot be built in combination with any of those I am unable to use Corrosion to build this package. 

I am at a bit of a loss of how to go about this as I would love to get this working and this package is a core dependency for many of my other crates. 

Any advice would be greatly appreciated!

**Comment by jschwe** [comment]
Could you elaborate on your problem? proc macro crates will be built perfectly fine as dependencies, since Corrosion invokes `cargo build`. 
Is there any reason you need to expose the proc macro lib as a CMake target?

**Reply by Christian-Prather** [reply]
Thanks for the reply,

We currently have a package callled `custom_macros` with a toml file that looks something like this 
```
[package]
name = "custom_macros"

[lib]
proc-macro = true

[dependencies]
syn = "2.0"
quote = "1.0"
proc-macro2 = "1.0"
darling = "0.20.1"
itertools = "0.11"
```
This crate is then itself a dependency for others in our workspace that I am trying to build with Corrosion. 
```
[dependencies]
async-trait = "0.1.68"
custom_macros = "*"
 ```

If there is something inherently wrong with this I am open to how it should be done. Thank you in advance!

**Reply by Christian-Prather** [reply]
I actually think I may have just been misunderstanding my errors. I believe the core issue is actually that Corrosion doesnt seem to be correctly picking up clippy allows and thus the proc-macros lib is panicking at compile time. I currently have this line in my custom_macros lib.rs `#![allow(clippy::expect_used, clippy::unwrap_used, clippy::panic)]` which was needed prior to my use of Corrosion but when I build with Corrosion I get errors around macros not used.
```
error: fields `fields` are never read
   |
23 | pub struct Variant {
   |            ------- fields in this struct
...
39 |     pub fields: proc_macro2::TokenStream,
   |         ^^^^^^
```

**Reply by jschwe** [reply, marked as answer]
Your error doesn't look like a clippy error and instead more like something that needs
`#[allow(dead_code)]` or similar.

**Reply by Christian-Prather** [reply]
Yes that was it thank you!
